crates/sparrow-instructions/src/evaluators/list/collect_boolean.rs

```diff
@@ -116,13 +116,13 @@ impl CollectBooleanEvaluator {
             let entity_index = *entity_index as usize;
 
             // Do not collect null values
-            if input.is_some() {
+            if let Some(input) = input {
                 self.token.add_value(self.max, entity_index, input);
             }
 
             let cur_list = self.token.state(entity_index);
             if cur_list.len() >= self.min {
-                list_builder.append_value(cur_list.iter().copied());
+                list_builder.append_value(cur_list.iter().map(|i| Some(*i)));
```

Comment:
Should I just leave `cur_list` as a `Vec<Option<T>>` so I don't have to re-wrap them in `Option` each iteration?